pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -0,0 +1,650 @@
+
+	if !ntconfig.InHyperShift() {
+		return reconcile.Result{}, fmt.Errorf("Using hypershift controller configuration while not in hypershift deployment")
+	}
+
+	instance := &corev1.ConfigMap{}
+
+	err := r.ManagementClient.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if instance.DeletionTimestamp != nil {
+		// ConfigMap is marked for deletion and waiting for finalizers to be empty
+		// so better to clean-up and delete the objects.
+
+		//REVIEW - Is it ok to use the ctx Context with the remote client???
+		if err := hypershiftDeleteComponents(r.Client, ctx, instance); err != nil {
+			klog.Errorf("failed to delete components: %v", err)
+			return reconcile.Result{}, err
+		}
+
+		// remove finalizer
+		if configMapHasFinalizer(instance, hypershiftFinalizer) {
+			cm := configMapRemoveFinalizer(instance, hypershiftFinalizer)
+			if err := r.ManagementClient.Update(ctx, cm); err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, nil
+		}
+	}
+
+	//add finalizer
+	if !configMapHasFinalizer(instance, hypershiftFinalizer) {
+		instance.Finalizers = append(instance.Finalizers, hypershiftFinalizer)
+		if err := r.ManagementClient.Update(ctx, instance); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	performanceProfileString, ok := instance.Data[tunedConfigMapConfigKey]
+	if !ok {
+		klog.Warning("ConfigMap %s has no data for field %s", instance.ObjectMeta.Name, tunedConfigMapConfigKey)
+
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	cmNodePoolNamespacedName, ok := instance.Annotations[hypershiftNodePoolLabel]
+	if !ok {
+		klog.Warningf("failed to parse PerformanceProfileManifests in ConfigMap %s, no label %s", instance.ObjectMeta.Name, hypershiftNodePoolLabel)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+	nodePoolName := parseNamespacedName(cmNodePoolNamespacedName)
+
+	performanceProfileFromConfigMap, err := parsePerformanceProfileManifest([]byte(performanceProfileString), nodePoolName)
+	if err != nil {
+		klog.Warningf("failed to parseTunedManifests in ConfigMap %s: %v", instance.ObjectMeta.Name, err)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	if err := updatePerformanceProfile(performanceProfileFromConfigMap, ctx, r.ManagementClient, instance, nodePoolName); err != nil {
+		klog.Errorf("failed to update performance profile from configMap %q components: %v", instance.Name, err)
+		//REVIEW - Should we record this events? and if so where? in the CM or the PP?
```

Comment:
For NTO we would just log such an error, I think.

Reply:
I have this questions because we have an EventRecorder as part of the PerformanceProfileReconciler in SNO that we use to record some of this failures ... and I do not know if this is applicable here. 

Anyway it could be added in a later PR if we think is needed.

WDYT? @yanirq, @fromanirh @Tal-or

Reply:
having events in the management cluster, if feasible, is nice. Let's try to be consistent with what classic PAO/PPCtrl is doing.

Reply:
I have added the events in one of the last commits.